demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -0,0 +1,370 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    Field,
+    FieldSet,
+    FinalForm,
+    FinalFormInput,
+    FinalFormSaveSplitButton,
+    FinalFormSubmitEvent,
+    Loading,
+    MainContent,
+    TextField,
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    ToolbarTitleItem,
+    useFormApiRef,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { ArrowLeft } from "@comet/admin-icons";
+import { EditPageLayout, queryUpdatedAt, resolveHasSaveConflict, useFormSaveConflict } from "@comet/cms-admin";
+import { IconButton } from "@mui/material";
+import { FormApi } from "final-form";
+import { filter } from "graphql-anywhere";
+import isEqual from "lodash.isequal";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createManufacturerMutation, manufacturerFormFragment, manufacturerQuery, updateManufacturerMutation } from "./ManufacturerForm.gql";
+import {
+    GQLCreateManufacturerMutation,
+    GQLCreateManufacturerMutationVariables,
+    GQLManufacturerFormDetailsFragment,
+    GQLManufacturerQuery,
+    GQLManufacturerQueryVariables,
+    GQLUpdateManufacturerMutation,
+    GQLUpdateManufacturerMutationVariables,
+} from "./ManufacturerForm.gql.generated";
+
+type FormValues = Omit<GQLManufacturerFormDetailsFragment, "address" | "addressAsEmbeddable"> & {
+    address: {
+        street: string;
+        streetNumber: string | null;
+        zip: string;
+        country: string;
+        alternativeAddress: { street: string; streetNumber: string | null; zip: string; country: string };
+    };
```

Comment:
For nested fields you should also just modify the ones that need it, like so:

```suggestion
    address: Omit<GQLManufacturerFormDetailsFragment["address"], "streetNumber"> & 
    {
        streetNumber: string | null;
        alternativeAddress: Omit<GQLManufacturerFormDetailsFragment["address"]["alternativeAddress"], "streetNumber"> &  { streetNumber: string | null; };
    };
```

(or try to use MergeDeep or something similar)

Reply:
solved 👍